language/tools/test-generation/src/transitions.rs

```diff
@@ -176,7 +176,7 @@ pub fn stack_satisfies_struct_signature(
         .flatten()
         .map(|field| field.type_signature().token());
     let mut satisfied = true;
-    for (i, token_view) in field_token_views.enumerate() {
+    for (i, token_view) in field_token_views.rev().enumerate() {
```

Comment:
Was this a bug in Darion's code where he got the order in which values are unpacked from the stack wrong?

Reply:
Yes. This was leading to us trying to pack in the types in the reverse order.